tests/mobile/common/common.specs.js

```diff
@@ -63,4 +65,71 @@ describe('Navigation tests : ', () => {
     expect(await display.isPresent()).toBeTruthy();
     await browser.get(utils.getBaseUrl() + 'messages/');
   });
+
+  //mobile resolution
+  describe('Mobile view tests : ', () => {
+    const district = {
+      _id: 'district_id',
+      type: 'clinic',
+      name: 'District',
+    };
+    const user = {
+      username: 'user',
+      password: 'Sup3rSecret!',
+      place:district._id,
+      contact: {
+        _id: 'some_id',
+        name: 'contact'
+      },
+      roles: ['program_officer']
+    };
+
+    const newPermissions =  {
+      can_view_messages: ['program_officer'],
+      can_view_messages_tab: ['program_officer'],
+      can_view_reports: ['program_officer'],
+      can_view_reports_tab: ['program_officer'],
+      can_view_contacts: ['program_officer'],
+      can_view_contacts_tab: ['program_officer'],
+      can_view_analytics: [],
+      can_view_nalytics_tab: []
+    };
+
+    let originalTimeout;
+
+    beforeEach(async () =>await utils.beforeEach());
+
+    beforeAll(async () => {
+      originalTimeout = jasmine.DEFAULT_TIMEOUT_INTERVAL;
+      jasmine.DEFAULT_TIMEOUT_INTERVAL = 2 * 60 * 1000;
+      await browser.driver.manage().window().setSize(389, 500);
+      await utils.saveDoc(district);
+      await utils.createUsers([user]);
+    });
+
+    afterAll(async () => {
+      jasmine.DEFAULT_TIMEOUT_INTERVAL = originalTimeout;
+      await browser.driver.manage().window().setSize(1024,768);
+      await utils.revertDb();
+    });
+
+    it('No tab text labels displayed  on mobile view for over 3 tabs', async () => {
+      const tabTexts = await element.all(by.css('.button-label')).getText();
+      expect(tabTexts.length).toBe(5);
+      expect(tabTexts).toEqual([ '', '', '', '', '' ]);
+    });
+
+    it('Display page tab text labels even on mobile view, whenever there are 3 or fewer tabs', async () => {
+      //change permissions
+      await utils.updateSettings({permissions:newPermissions});
+      await commonElements.goToLoginPageNative();
+      await loginPage.loginNative(user.username, user.password);
+      await commonElements.waitForLoaderToDisappear(200000);
```

Comment:
I think this is an extreme timeout. It also looks like all tests that follow this one fail with an error about not being able to load settings. 
I think you need to do this in the afterAll of this test: 
1. Delete the user (with utis.deleteUsers)
2. Logout and login as admin